lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -121,28 +109,62 @@ def self.process_result(ae_result, aevent)
   rescue URI::InvalidURIError => err
   end
 
-  def self.automate_user
-    # TODO: Get the real userid to pass into Automate
-    user = User.where(:userid => "admin").first
-    raise "Admin user needed to raise events" unless user
-    user
-  end
-
-  def self.call_automate(obj_type, obj_id, attrs, instance_name,
-                         user, state = nil, message = nil)
+  def self.call_automate(obj, attrs, instance_name, message = nil)
+    user_id, group_id, tenant_id = automate_user_ids(obj)
+    attrs.merge!(
+      :user_id      => user_id,
+      :miq_group_id => group_id,
+      :tenant_id    => tenant_id
+    )
     args = {
-      :object_type      => obj_type,
-      :object_id        => obj_id,
+      :object_type      => obj.class.name,
+      :object_id        => obj.id,
       :attrs            => attrs,
       :instance_name    => instance_name,
-      :user_id          => user.id,
-      :miq_group_id     => user.current_group.id,
-      :tenant_id        => user.current_tenant.id,
-      :automate_message => message,
-      :state            => state
+      :user_id          => user_id,
```

Comment:
@lfu @kbrock 
The user_id, mi_group_id, tenant_id should be part of args and not part of attrs. We don't look at attrs to get the miq_group_id and tenant_id.